ncm-cron/src/main/pan/components/cron/schema.pan

```diff
@@ -38,12 +38,63 @@ type structure_cron_log = {
     'mode' ? string
 } with structure_cron_log_valid(SELF);
 
+@documentation{
+    Validate contents of cron fields
+}
+function valid_cron_field = {
+    if (ARGC == 4) {
+        test_value = ARGV[0];
+        lower_bound = ARGV[1];
+        upper_bound = ARGV[2];
+        text_regex = ARGV[3];
+
+        # Check that the field contains only valid characters
+        result = match(test_value, '^(?:[a-z0-9/*-,]+)$');
+        if (!result) error(format('"%s" contains invalid characters', test_value));
+
+        # Find all strings of digits and validate them against provided bounds
+        foreach(k; v; matches(test_value, '([0-9]+)')) {
+            i = to_long(v);
+            if (i < lower_bound) error(format('Value %d is below lower bound of %d', i, lower_bound));
+            if (i > upper_bound) error(format('Value %d is above upper bound of %d', i, upper_bound));
+            result = result && i >= lower_bound && i <= upper_bound;
+        };
+
+        # Find all strings of letters and validate them against provided regex
+        foreach(k; v; matches(test_value, '([a-z]+)')) {
+            result = result && match(v, text_regex);
+            if (!result) error(format('"%s" is not a valid value for this field item', v));
+        };
+
+        # Ignore all other characters
+        return(result);
+    } else {
+        error(format('valid_cron_field_items: expected 4 parameters, received %d', ARGC));
```

Comment:
i prefer a `if (ARGC != 4) error(...)` in the beginning to avoid an extra indentation
also, you can use builtin `FUNCTION` with format for the function name